File: lib_base/src/main/java/com/luckyaf/smartandroid/lib/base/utils/ToastUtil.java
package com.luckyaf.smartandroid.lib.base.utils;

import android.content.Context;
import android.widget.Toast;

/**
 * 类描述：
 *
 * @author Created by luckyAF on 2018/6/29
 */
@SuppressWarnings("unused")
public class ToastUtil {
    private static Toast toast;

    private ToastUtil() {
        throw new UnsupportedOperationException("cannot be instantiated");
    }


    public static void showMessage(Context context, CharSequence message) {
        if (toast != null) {
            toast.setText(message);
            toast.show();
        } else {
            toast = Toast.makeText(context, message, Toast.LENGTH_SHORT);
            toast.show();
        }
    }

    public static void showMessage(Context context, int message) {
        if (toast != null) {
            toast.setText(message);
            toast.show();
        } else {
            toast = Toast.makeText(context, message, Toast.LENGTH_SHORT);
            toast.show();
        }
    }

}

File: module_kotlin/src/main/java/com/luckyaf/oneforall/kotlin/utils/ToastUtil.kt
package com.luckyaf.oneforall.kotlin.utils

import android.content.Context
import android.widget.Toast

/**
 * 类描述：
 * @author Created by luckyAF on 2018/7/18
 *
 */
object ToastUtil {

    private var  toast : Toast ?= null
    fun showMessage(context: Context?,message : CharSequence){
        if(null == toast){
            toast = Toast.makeText(context,message,Toast.LENGTH_SHORT)
        }else{
            toast?.setText(message)
        }
        toast?.show()
    }
    fun showMessage(context: Context?,resourceId:Int){
        if(null == toast){
            toast = Toast.makeText(context,resourceId,Toast.LENGTH_SHORT)
        }else{
            toast?.setText(resourceId)
        }
        toast?.show()
    }
}